### PR TITLE
Enable cleanup in E2E

### LINKED ISSUE #522: [E2E Enchancement] define command line flags for different cleanup granularity
**What would you like to be added**:

Implement more command line flags for different cleanup granularity to make it more convenient for local run and debug flaky test case.

Now after running `test/e2e/run_test.sh`, infra like k8s, istio and Kmesh will not cleanup, but test applications will.

So add command line flag to implement such as:

1. Cleanup all components: this seems simple, just use `kind` to delete the testing k8s cluster.
2. Don't cleanup test applications: we can use `istio.test.nocleanup` which is the flag of istio integration framework to control it.

And more if it's necessary.

### PR DESCRIPTION
**Pull request by noobwei**
**What type of PR is this?**

/kind enhancement

**What this PR does / why we need it**:

Enable cleanup in E2E, include cleanup the kind cluster or docker registry or both

**Which issue(s) this PR fixes**:
Fixes #522 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: test/e2e/run_test.sh [hunk 1]
```diff
@@ -208,6 +225,19 @@ while (( "$#" )); do
...
     ;;
+    --cleanup-kind)
+      CLEANUP_KIND=true
+      shift
+    ;;
+    --cleanup-registry)
+      CLEANUP_REGISTRY=true
+      shift
+    ;;
+    --cleanup-all)
```
Comment (hzxuzhonghu) [inline test/e2e/run_test.sh:236]: I think cleanup both kind and dockerregistry is meaningful, not only one of them

File: test/e2e/run_test.sh [hunk 2]
```diff
@@ -184,6 +184,23 @@ function install_dependencies() {
...
 
+function cleanup_kind_cluster() {
+    local NAME="${1:-kmesh-testing}"
+    echo "Deleting KinD cluster with name=${NAME}"
+    kind delete cluster --name="${NAME}"
+    echo "KinD cluster ${NAME} cleaned up"
+}
+
+function cleanup_docker_registry() {
+    local REG_NAME="${KIND_REGISTRY_NAME:-kind-registry}"
```
Comment (YaoZengzeng) [inline test/e2e/run_test.sh:195]: We could use `KIND_REGISTRY_NAME` directly, it couldn't be configured now.

**Comment by YaoZengzeng** [comment]
@noobwei should rebase first, as I've modified `test/e2e/run_test.sh`.

**Comment by YaoZengzeng** [comment]
Actually you don't need to close this PR, just use "git push -f" to force overwrite.

**Pull request closed without merging**